### ISSUE TITLE
Benchmark cost/benefit of using large pages for linear mapping

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#367]
For Book3S 64 we have used 16MB pages for mapping the linear mapping since the original port was merged.

With the addition of the Radix MMU we now also use 2MB/1GB pages.

This provides _obvious_ benefits in terms of reduced pressure on the HPT, or less memory used by page tables for Radix, as well as reduced TLB usage.

However we don't have any solid benchmarks to back that up, because we've always done it that way. There is increasing interest in using PAGE_SIZE pages for the linear mapping, to support various hardening/security use cases. So it would be good to have some numbers to help us understand what the trade offs are.

Some work has been done by Intel on x86:
  https://lore.kernel.org/linux-mm/213b4567-46ce-f116-9cdf-bbd0c884eb3c@linux.intel.com

They compare 4K/2MB/1G, and obviously the x86 MMU is different to ours. But it's possible the big picture would be similar for Radix. One of their conclusions is that 1G pages give more benefit as total RAM size increases, although they only tested with small amounts of memory, up to 512GB in total.

So tldr we would like some benchmark numbers for using/not-using large pages for the linear mapping, across a range of benchmarks and memory sizes.

**Comment by rppt** [comment]
I've run several tests from [mmtest](https://github.com/gormanm/mmtests) suite on a P8 machine (8348-21C with 8 cores x 8 threads and 128G RAM) with hash mmu. I've forced the linear map page size to 1M or 64K and used 16M pages as the baseline.
Most of the time the differences in benchmark performance are really small. There are some exceptions, but I believe these are spikes that would be flattened out with more repetitions.

The raw results are here: https://gist.github.com/rppt/e8b58135a2e06c6997e12fb30757e785
